pkg/portfwd/forward_test.go

```diff
@@ -0,0 +1,148 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package portfwd
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/lima-vm/lima/v2/pkg/guestagent/api"
+	"github.com/lima-vm/lima/v2/pkg/limatype"
+)
+
+func TestForwarderIgnoreSkipsDynamicListeners(t *testing.T) {
+	dial := func(context.Context, string, string) (net.Conn, error) {
+		return nil, errors.New("unexpected dial")
+	}
+
+	guestIP := net.ParseIP("127.0.0.1")
+	hostIP := net.ParseIP("127.0.0.1")
+	guestRange := [2]int{1, 65535}
+	falseVal := false
+	rules := []limatype.PortForward{{
+		Proto:             limatype.ProtoAny,
+		GuestIPMustBeZero: &falseVal,
+		GuestIP:           guestIP,
+		GuestPortRange:    guestRange,
+		HostIP:            hostIP,
+		HostPortRange:     guestRange,
+	}}
+
+	tests := []struct {
+		name      string
+		protocol  string
+		ignoreTCP bool
+		ignoreUDP bool
+	}{
+		{name: "tcp ignored", protocol: "tcp", ignoreTCP: true},
+		{name: "udp ignored", protocol: "udp", ignoreUDP: true},
+	}
+
+	for _, tt := range tests {
+		tc := tt
+		t.Run(tc.name, func(t *testing.T) {
+			preCheck := NewPortForwarder(rules, false, false)
+			defer preCheck.Close()
+			port := &api.IPPort{Protocol: tc.protocol, Ip: "127.0.0.1", Port: 23456}
+			local, _ := preCheck.forwardingAddresses(port)
+			assert.Assert(t, local != "", "test precondition failed: expected forwarding address for %s", tc.protocol)
```

Comment:
Why check 2 portForwarders for every case? We seem to do the same check (false, false) for every test.